design/pupil-state.md

```diff
@@ -3,6 +3,19 @@
 With the introduction of the pupilState table we will have to keep this up to date with the latest
 references when related events occur.  For example - when a pin is generated for a pupil it creates a check record.  This will require the check_id being inserted into the pupilState table.
 
+## States required
+
+- can take check: can have a pin & check activated by a teacher
+- taking check: pin & check allocated
+- check received: submitted check received, awaiting processing
+- complete: check is valid and marked.  no further action necessary
+- processing failed: submitted check could not be processed
+- restart available: teacher has activated restart for pupil, not yet consumed
+- restart consumed: pupil logged in and collected the check made available by restart
+  - is this essential? could we not allow a restart even if not logged in? as recording login to sql is expensive
+- not attending: pupil will not be taking the check
```

Comment:
Seems like Check not received is missing?

Reply:
i'm working on this right now, off the back of the conversation we had in the restarts channel.

effectively we need an 'expired' flag

Reply:
Expired is separate to Not Received.  Not receives requires a restart.

Reply:
yes.  the expiry service needs to accommodate the fact that submitted checks may be in a queue waiting to be acknowledged as received.   The initial idea is to wait for until the next day, and if it still has not been marked as received, then it should be expired.  We can never be sure whether they did in fact submit, or just closed the check.  See `design/pupil-state.md` for notes on this situation.